### PR TITLE
support max member limit on roles and groups

### PR DESCRIPTION
**Pull request by havetisyan**
# Description
support max member limit on roles and groups

# Contribution Checklist:
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have read the [contribution guidelines](https://github.com/AthenZ/athenz/blob/master/CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request.**

### CODE REVIEW

File: servers/zms/schema/zms_server.sql [hunk 1]
```diff
@@ -90,6 +90,7 @@ CREATE TABLE IF NOT EXISTS `zms_server`.`role` (
   `group_expiry_days` INT NOT NULL DEFAULT 0,
   `description` VARCHAR(4096) NOT NULL DEFAULT '',
   `delete_protection` TINYINT(1) NOT NULL DEFAULT 0,
+  `max_members` INT NOT NULL DEFAULT 0,
```
Comment (MartinTrojans) [inline servers/zms/schema/zms_server.sql:93]: Should we have some other numbers as the default value instead of 0?

Comment (havetisyan) [reply servers/zms/schema/zms_server.sql:93]: No, we can't introduce backward incompatible changes. Value of 0 indicates there is no limit which is the current behavior.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/QuotaChecker.java [hunk 2]
```diff
@@ -178,49 +194,93 @@ void checkGroupQuota(ObjectStoreConnection con, String domainName, Group group,
...
+
+        Membership membership = con.getRoleMember(domainName, roleName, memberName, 0, false);
+        if (membership.getIsMember()) {
+            return;
+        }
+
+        // so at this point we know that we'll be adding a new member to the
+        // role. so first let's count the number of role members
+
+        int roleMemberCount = con.countRoleMembers(domainName, roleName);
```
Comment (chandrasekhar1996) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/QuotaChecker.java:220]: Is it meant to count the expired members also as valid role/group member in terms of count?

Comment (havetisyan) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/QuotaChecker.java:220]: yes, the count includes everyone regardless if they're expired or not

**Review by chandrasekhar1996 — APPROVED** [review]

**Pull request merged**